### PR TITLE
TST: Use templates in Azure Pipelines config file

### LINKED ISSUE #928: Clean up azure-piplines.yml with templates
**Is your feature request related to a problem? Please describe.**
It would be more ideal for the Azure Pipelines (AP) config file to be more clearly readable and similar platforms treated identically.

**Describe the solution you'd like**
Implementing templates in AP would accomplish the above.

**Describe alternatives you've considered**
No other good alternatives exist to accomplish this goal.

### PR DESCRIPTION
**Pull request by CameronTStark**
- [x] Closes #928
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

The aim is to simplify the format of the azure-pipelines.yml config file and standardize tests as much as possible across versions.

### CODE REVIEW

**Comment by CameronTStark** [comment]
This should be ready to review at your convenience despite the codecov nuisance-failure.

File: ci/azure/conda_linux.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    displayName: 'List installed dependencies'
+  - script: |
+      source activate test_env
+      export NREL_API_KEY=$(nrelApiKey)
+      pytest pvlib --remote-data --junitxml=junit/test-results.xml --cov --cov-report=xml --cov-report=html
+    displayName: 'pytest'
+  # - script: |
+  #     source activate test_env
+  #     flake8 pvlib
+  #   displayName: 'flake8'
```
Comment (wholmgren) [inline ci/azure/conda_linux.yml]: delete?

Comment (CameronTStark) [reply ci/azure/conda_linux.yml]: Done. I was wondering if there was a reason it was left there.

**Comment by mikofski** [comment]
Thanks Cameron, much cleaner.

**Comment by wholmgren** [comment]
thanks @CameronTStark !

**Pull request merged**